docs/2.0/docs/pipelines/guides/file-dependencies.md

```diff
@@ -0,0 +1,15 @@
+# Running Pipelines for Changes to non-IaC file dependencies
+
+Pipelines runs plan/applies based on changes to your units in your IaC repository. Intuitively, if you make a change to `path/to/my/unit/terragrunt.hcl` then Pipelines will run plan/apply in `/path/to/my/unit`.  Pipelines also supports running `run-all` when [consolidation](/2.0/reference/pipelines/configurations-as-code/api#consolidate_added_or_changed) is enabled.
+
+However, there are some cases where you may want to run Pipelines for changes to files that are not part of your IaC. For example, you may have a `terragrunt.hcl` file that imports data from a `tags.yaml` file. If you make a change to the `tags.yaml` file, you may want to run Pipelines to plan/apply the changes to the associated `terragrunt.hcl` file.  This page documents how to configure pipelines to automatically run `plan/apply` when only these data files are changed.
+
+## Configuration
+
+Pipelines will automatically detect, based on git commit history, when data files have changed and emit a `FileChanged` event. This will create a new job in your workflow that will invoke Terragrunt with the [`queue-include-units-reading`](https://terragrunt.gruntwork.io/docs/reference/cli-options/#terragrunt-queue-include-units-reading) option set to the files that have changed. Terragrunt will then scan your code for units that depend on the changed files and add them to the plan/apply.
```

Comment:
What does "emit a `FileChanged` event" actually mean? What exactly emits that event? How is it picked up? Likewise, when you say "new job" and "workflow", what do you mean?

I realize that's not the focus of these docs, but as someone not super familiar with the technical internals of Pipelines, I was thrown off a little by this.

Also, I'm realizing that we're using some GitHub Actions-specific language here (job, workflow). Should we define general terms for these, and link to the official definitions so that users can know what we mean? Or perhaps say "GitHub Actions job" and "GitHub Actions workflow"?